src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -416,6 +416,30 @@ public BoundBlock Block(ImmutableArray<LocalSymbol> locals, ImmutableArray<Bound
             return new BoundBlock(Syntax, locals, statements) { WasCompilerGenerated = true };
         }
 
+        public BoundBlock StateMachineScope(ImmutableArray<StateMachineFieldSymbol> hoistedLocals, BoundStatement statement)
+        {
+            return Block(new BoundStateMachineScope(Syntax, hoistedLocals, statement));
+        }
+
+        public bool TryUnwrapBoundStateMachineScope(ref BoundStatement statement, out ImmutableArray<StateMachineFieldSymbol> hoistedLocals)
```

Comment:
This feels too specialized to be in the Factory. Consider moving to AsyncMethodToStateMachineRewriter.cs instead. And does it depend on the Factory instance?